packages/context/src/context.ts

```diff
@@ -21,12 +28,33 @@ export class Context {
    * Name of the context
    */
   readonly name: string;
+
+  /**
+   * Key to binding map as the internal registry
+   */
   protected readonly registry: Map<string, Binding> = new Map();
+
+  /**
+   * Parent context
+   */
   protected _parent?: Context;
 
   /**
-   * Create a new context
+   * A list of registered context listeners
+   */
+  public readonly listeners: Set<ContextEventListener> = new Set();
+
+  /**
+   * Create a new context. For example,
+   * ```ts
+   * const ctx1 = new Context();
+   * const ctx2 = new Context(ctx1);
+   * const ctx3 = new Context('ctx3');
+   * const ctx4 = new Context(ctx3, 'ctx4');
```

Comment:
I find these examples confusing, because it's not clear why the parameters are passed as strings (`'ctx3'`,  `'ctx4'`) in some place but as context instances (`ctx1`, `ctx3`) in other places.

Please use more descriptive names in the string arguments, e.g.

```ts
const ctx1 = new Context();
const ctx2 = new Context(ctx1);
const ctx3 = new Context('another root context');
const ctx4 = new Context(ctx3, 'a child context');
```

Or even better, add a short comment explaining each form.

```ts
// Create a new root context, let the framework to create a unique name
const ctx1 = new Context();

// Create a new child context inheriting bindings fro ctx1
const ctx2 = new Context(ctx1);

// Create another root context called "application"
const ctx3 = new Context('application');

// Create a new child context called "request" and inheriting bindings from ctx3
const ctx4 = new Context(ctx3, 'request');
```

Reply:
_reposting an older comment: https://github.com/strongloop/loopback-next/pull/2122/files#r245919522_

I find these examples confusing, because it's not clear why the parameters are passed as strings (`'ctx3'`,  `'ctx4'`) in some place but as context instances (`ctx1`, `ctx3`) in other places.

Please use more descriptive names in the string arguments, e.g.

```ts
const ctx1 = new Context();
const ctx2 = new Context(ctx1);
const ctx3 = new Context('another root context');
const ctx4 = new Context(ctx3, 'a child context');
```

Or even better, add a short comment explaining each form.

```ts
// Create a new root context, let the framework to create a unique name
const ctx1 = new Context();

// Create a new child context inheriting bindings fro ctx1
const ctx2 = new Context(ctx1);

// Create another root context called "application"
const ctx3 = new Context('application');

// Create a new child context called "request" and inheriting bindings from ctx3
const ctx4 = new Context(ctx3, 'request');
```